plugins/org.eclipse.n4js/src/org/eclipse/n4js/generator/ISubGenerator.java

```diff
@@ -49,4 +50,9 @@
 	 * @return the compiled result as String
 	 */
 	public String getCompileResultAsText(Script root, GeneratorOption[] options);
+
+	/**
+	 * Return the set of applicable file extensions.
+	 */
+	public Set<String> applicableFileExtensions(Resource input);
```

Comment:
would be cleaner to provide this information as part of the extension in a plugin.xml